### ISSUE TITLE
Remove unittest code from non-test code

### ISSUE THREAD

**Issue by mwatts15** [gitpython-developers/GitPython#790]
I see these imports of unittest in
- https://github.com/gitpython-developers/GitPython/blob/a8591a094a768d73e6efb5a698f74d354c989291/git/objects/submodule/base.py#L6
- https://github.com/gitpython-developers/GitPython/blob/f3265bd8beb017890699d093586126ff8af4a3fe/git/util.py#L17

I generally think that test code should not be built into the business logic. Is there a reason why using a skipIf decorator is insufficient?

**Comment by Byron** [comment]
Good catch, I absolutely agree! Even though I am pretty sure there was a very good reason to do that at some point in time. With the current state of windows test, for which these dependencies seemingly have been introduced. Maybe @ankostis can provide more context.
